options/posix/generic/net-if-stubs.cpp

```diff
@@ -10,9 +10,24 @@ void if_freenameindex(struct if_nameindex *idx) {
 	free(idx);
 }
 
-char *if_indextoname(unsigned int, char *) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+char *if_indextoname(unsigned int index, char *name) {
+	struct ifreq ifr;
+	int fd, r;
+
+	if((fd = socket(AF_UNIX, SOCK_DGRAM|SOCK_CLOEXEC, 0)) < 0) {
+		return 0;
+	}
+
+	ifr.ifr_ifindex = index;
+	r = ioctl(fd, SIOCGIFNAME, &ifr);
+	close(fd);
+	if (r < 0) {
+		if(errno == ENODEV) {
+			errno = ENXIO;
+		}
+		return 0;
+	}
+	return strncpy(name, ifr.ifr_name, IF_NAMESIZE);
```

Comment:
I think this should be a sysdep, this is the Linux implementation, others might not just pick a random socket to run an ioctl on to resolve interface names.